plugins/linux/ifplugin/linuxcalls/netlink_api.go

```diff
@@ -80,13 +125,40 @@ type NetlinkAPIRead interface {
 	// GetChecksumOffloading returns the state of Rx/Tx checksum offloading
 	// for the given interface.
 	GetChecksumOffloading(ifName string) (rxOn, txOn bool, err error)
+	// DumpInterfaces uses local cache to gather information about linux
+	// namespaces and retrieves interfaces from them.
+	DumpInterfaces() ([]*InterfaceDetails, error)
+	// DumpInterfacesWithContext retrieves all linux interfaces based
+	// on provided namespace context.
+	DumpInterfacesWithContext(nsList []*namespaces.NetNamespace) ([]*InterfaceDetails, error)
```

Comment:
Or what about `DumpInterfacesFromNamespaces`? `WithContext` is often used for functions that accept `ctx context.Context`  argument.

Reply:
There was a context in an earlier implementation, hence the name. Later I've changed it to a simple namespace list so I agree, it should be renamed.